sdk/src/lib.rs

```diff
@@ -105,3 +105,123 @@ extern crate log as logger;
+///     * `desc` - human-readable description of the account for documentation, name-value format: `desc = "<description>"`
+///   * `skip` - variant will be skipped in helper-function and verbose-enum expansion
+///
+///
+/// Example input enum:
+///
+/// ```rust
+/// use bincode::serialize;
+/// use serde_derive::Serialize;
+/// use solana_sdk::instructions;
+/// mod test_program {
+///     solana_sdk::declare_id!("8dGutFWpfHymgGDV6is389USqGRqSfpGZyhBrF1VPWDg");
+/// }
+///
+/// #[instructions(test_program::id())]
+/// #[derive(Clone, Debug, PartialEq, Serialize)]
+/// pub enum TestInstruction {
+///     /// Transfer lamports
+///     #[accounts(
+///         from_account(SIGNER, WRITABLE, desc = "Funding account"),
+///         to_account(WRITABLE, desc = "Recipient account")
+///     )]
+///     Transfer {
+///         /// The transfer amount
+///         lamports: u64,
+///     },
+/// }
+/// ```
+///
+///
+/// Example output:
+///
+/// ```rust
+/// use bincode::serialize;
+/// use serde_derive::Serialize;
+/// use solana_sdk::{
+///     instruction::{AccountMeta, Instruction},
+///     pubkey::Pubkey,
+/// };
+/// mod test_program {
+///     solana_sdk::declare_id!("8dGutFWpfHymgGDV6is389USqGRqSfpGZyhBrF1VPWDg");
+/// }
+///
+/// #[derive(Clone, Debug, PartialEq, Serialize)]
+/// pub enum TestInstruction {
+///     /// Transfer lamports
+///     #[doc = "<br/>"]
+///     #[doc = "* Accounts expected by this instruction:"]
+///     #[doc = "  0. `[WRITABLE, SIGNER]` Funding account"]
+///     #[doc = "  1. `[WRITABLE]` Recipient account"]
+///     Transfer {
+///         /// The transfer amount
+///         lamports: u64
+///     },
+/// }
+///
+/// pub fn transfer(
+///     from_account: Pubkey,
+///     to_account: Pubkey,
+///     data: Vec<u8>,
+/// ) -> Instruction {
+///     let mut accounts: Vec<AccountMeta> = vec![];
+///     accounts.push(AccountMeta::new(from_account, true));
+///     accounts.push(AccountMeta::new(to_account, false));
+///
+///     Instruction {
+///         program_id: test_program::id(),
+///         data,
+///         accounts,
+///     }
+/// }
+///
+/// #[derive(Clone, Debug, PartialEq, Serialize)]
+/// pub enum TestInstructionVerbose {
+///     /// Transfer lamports
+///     Transfer {
+///         from_account: u8,
+///         to_account: u8,
+///         /// The transfer amount
+///         lamports: u64
+///     },
+/// }
+///
+/// impl TestInstructionVerbose {
+///    pub fn from_instruction(instruction: TestInstruction, account_keys: Vec<u8>) -> Self {
+///        match instruction {
+///            TestInstruction::Transfer { lamports } => TestInstructionVerbose::Transfer {
+///                from_account: account_keys[0],
+///                to_account: account_keys[1],
+///                lamports,
+///             }
+///         }
+///     }
+/// }
+/// ```
+pub use solana_sdk_program_macros::instructions;
```

Comment:
Is this file missing from the PR?

Reply:
This is the macro

Reply:
I'm missing something. I see big comment block and an `pub use`, but no `solana-sdk-program-macros/src/lib.rs`.

Reply:
nevermind, found it. So big GitHub collapsed it and I went by it like it were a Cargo.lock file